### PR TITLE
Added first-cut implementation of XR layers

### PR DESCRIPTION
**Pull request by asajeffrey**
Starting on an implementation of https://immersive-web.github.io/layers/

This PR doesn't add any new functionality, just the plumbing.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] There are tests for these changes OR

### CODE REVIEW

**Comment by asajeffrey** [comment]
It's still a draft because it needs some tests added.

**Comment by asajeffrey** [comment]
@Manishearth r?

File: components/script/dom/xrrenderstate.rs [hunk 1]
```diff
@@ -78,6 +126,24 @@ impl XRRenderState {
...
+    }
+    pub fn borrow_layers(&self) -> Layers {
+        Layers(self.layers.borrow())
+    }
+    pub fn has_layer(&self) -> bool {
+        self.layer.get().is_some() || !self.layers.borrow().is_empty()
+    }
+}
+
+pub struct Layers<'a>(Ref<'a, Vec<XRWebGLLayerOrXRLayer>>);
```
Comment (Manishearth) [inline components/script/dom/xrrenderstate.rs:140]: i typically prefer creating a `with_layers` method, see `with_session`. but no strong preference.

**Review by Manishearth — APPROVED** [review]

File: components/script/dom/xrrenderstate.rs (hunk 1 above)
Comment (asajeffrey) [reply components/script/dom/xrrenderstate.rs]: Done.

**Comment by asajeffrey** [comment]
@bors-servo retry
 - #25533

**Pull request merged**